StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Saved Payment Method Screen/Vertical Saved Payment Method Screen/PaymentMethodRowButton.swift

```diff
@@ -92,78 +94,58 @@ final class PaymentMethodRowButton: UIView {
         return circleView
     }()
 
-    lazy var removeButton: CircularButton = {
+    private lazy var removeButton: CircularButton = {
         let removeButton = CircularButton(style: .remove, iconColor: .white)
         removeButton.backgroundColor = appearance.colors.danger
         removeButton.isHidden = true
         removeButton.addTarget(self, action: #selector(handleRemoveButtonTapped), for: .touchUpInside)
         return removeButton
     }()
 
-    private lazy var editButton: CircularButton = {
-        let editButton = CircularButton(style: .edit, iconColor: .white)
-        editButton.backgroundColor = appearance.colors.icon
-        editButton.isHidden = true
-        editButton.addTarget(self, action: #selector(handleEditButtonTapped), for: .touchUpInside)
-        return editButton
+    private lazy var updateButton: CircularButton = {
+        let updateButton = CircularButton(style: .edit, iconColor: .white)
+        updateButton.backgroundColor = appearance.colors.icon
+        updateButton.isHidden = true
+        updateButton.addTarget(self, action: #selector(handleUpdateButtonTapped), for: .touchUpInside)
+        return updateButton
     }()
 
     private lazy var stackView: UIStackView = {
-        return UIStackView.makeRowButtonContentStackView(arrangedSubviews: [paymentMethodImageView, label, .makeSpacerView(), circleView, editButton, removeButton])
+        return UIStackView.makeRowButtonContentStackView(arrangedSubviews: [.makeSpacerView(), circleView, updateButton, removeButton])
     }()
 
-    private lazy var shadowRoundedRect: ShadowedRoundedRectangle = {
-        let shadowRoundedRect = ShadowedRoundedRectangle(appearance: appearance)
-        shadowRoundedRect.translatesAutoresizingMaskIntoConstraints = false
-        shadowRoundedRect.addAndPinSubview(stackView)
-        return shadowRoundedRect
-    }()
+    private lazy var rowButton: RowButton = {
+        let button: RowButton = .makeForSavedPaymentMethod(paymentMethod: paymentMethod, appearance: appearance) { [weak self] _ in
+            guard let self, !isEditing else { return }
+            state = .selected
+            delegate?.didSelectButton(self, with: paymentMethod)
+        }
 
-    private lazy var selectionTapGesture: UITapGestureRecognizer = {
-        return UITapGestureRecognizer(target: self, action: #selector(handleSelectionTap))
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.addAndPinSubview(stackView)
```

Comment:
I don't think this is right, this means the button content can overlap the `stackView` here?  `RowButton` has an unimplemented `rightAccessoryView` that I was thinking we would use for this purpose.

Reply:
Ah, thanks. Didn't see `rightAccessoryView`, updated to use it.